src/mds/nameserver2/clean_manager.cpp

```diff
@@ -60,9 +64,24 @@ bool CleanManager::SubmitDeleteCommonFileJob(const FileInfo &fileInfo) {
 bool CleanManager::SubmitCleanDiscardSegmentJob(
     const std::string& cleanSegmentKey,
     const DiscardSegmentInfo& discardSegmentInfo) {
+    // get dlock
+    dlockOpts_->pfx = std::to_string(discardSegmentInfo.fileinfo().id());
+    dlock_ = std::make_shared<DLock>(*dlockOpts_);
+    if (0 == dlock_->Init()) {
+        LOG(ERROR) << "Init DLock error"
+                << ", pfx = " << dlockOpts_->pfx
+                << ", retryTimes = " << dlockOpts_->retryTimes
+                << ", ctx_timeoutMS = " << dlockOpts_->ctx_timeoutMS
+                << ", ttlSec = " << dlockOpts_->ttlSec;
+        return false;
+    }
+    if (EtcdErrCode::EtcdOK != dlock_->Lock()) {
+        return false;
+    }
     auto task = std::make_shared<SegmentCleanTask>(
         cleanCore_, cleanSegmentKey, discardSegmentInfo);
     task->SetTaskID(reinterpret_cast<TaskIDType>(task.get()));
+    task->SetDLock(dlock_);
```

Comment:
you can pass `dlock_` in SegmentCleanTasks' constructor

Reply:
> you can pass `dlock_` in SegmentCleanTasks' constructor
done